enhancements/node-tuning/mixed-cpu-node-plugin.md

```diff
@@ -0,0 +1,316 @@
+the container-runtime configuration.
+
+The plugin is completely optional and is off by default.
+
+No changes are needed in the container-runtime, because NRI hooks already supported by the runtime
+and mixed-cpu-node-plugin implementing the NRI interface.
+This minimizes risks and helps avoid regression on
+a container-runtime component (CRI-O in OpenShift case).
+
+### API
+
+At first, the feature is enabled via annotation in the PerformanceProfile:
+
+```yaml
+annotations:
+    mixed-cpu.openshift.io/shared-cpus: "2-3"
+```
+
+The CPU set value should follow the 
+[cpuset](https://github.com/openshift/cluster-node-tuning-operator/blob/master/docs/performanceprofile/performance_profile.md#cpuset) conventions.
+It defines a set of CPUs that can be allocated to guaranteed pods/containers that still require
+non-exclusive ones.
+
+The CPU set must not overlap with `spec.cpu.reserved` or `spec.cpu.isolated` and NTO
+should return an error in case it does.
+
+The reason for using an annotation and not new API field 
+is because there is a coming feature named shared-partition
+that uses similar convention i.e., shared cpus, and would be 
+enabled via NTO as well.
+
+Once shared-partition lands, the annotation should be deprecated
+and node-plugin will use the shared cpus specified 
+at `spec.cpu.shared`.
+
+In case both annotation and `spec.cpu.shared` are specified, 
+the value at `spec.cpu.shared` has precedence over the annotation.
+
+PerformanceProfile example:
+```yaml
+apiVersion: performance.openshift.io/v2
+kind: PerformanceProfile
+metadata:
+  name: example-performance-profile
+  annotations:
+    mixed-cpu.openshift.io/shared-cpus: "2-3"
+spec:
+  cpu:
+    reserved: "0-1"
+    isolated: "4-8"
+  hugepages:
+    defaultHugepagesSize: "1G"
+    pages:
+      - size: "1G"
+        count: 2
+        node: 0
+  realTimeKernel:
+    enabled: true
+  nodeSelector:
+    node-role.kubernetes.io/performance: "test"
+```
+
+Specify the annotation activates the feature, signals NTO to update the `reservedSystemCpus` in Kubelet config, 
+and deploys the mixed-cpu-node-plugin.
+The logic of updating reserved-cpus pool is NTO's responsibility and would be implemented as part of this feature.
+
+Remove the annotation reverts the changes from Kubelet `reservedSystemCpus`
+and deletes the mixed-cpu-node-plugin from the cluster.
+
+NOTE: After shared-partition lands, the shared cpus will be allocated from `spec.cpu.shared` cpuset
+and no longer be allocated from Kubelet `reservedSystemCpus` cpuset.
+
+### Workflow Description
+
+The premise is that OCP cluster is running, and there's an active performance-profile that already tuned the system.
+
+1. The cluster administrator wants to support both exclusive and shared cpus for workloads running on the cluster.
+2. The cluster administrator specifies the shared CPU ids via annotation in the PerformanceProfile's.
+3. The cluster administrator waits for MCO to kick in and update the nodes.
```

Comment:
I assume this triggers a reboot? I think that should be mentioned.

Reply:
I will thanks